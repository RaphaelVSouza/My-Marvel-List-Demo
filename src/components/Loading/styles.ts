import styled, { keyframes } from 'styled-components'

const rotate = keyframes`
    from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`

export const LoadingIcon = styled.img`
  animation: ${rotate} 2s linear infinite;
`
