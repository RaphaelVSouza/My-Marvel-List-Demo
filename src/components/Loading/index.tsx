import * as S from './styles'

const Loading = () => (
  <S.Wrapper>
    <S.LoadingIcon
      src={`${process.env.PUBLIC_URL}/loading-icon.svg`}
      alt="a circle spinning representing a loading"
    />
  </S.Wrapper>
)

export default Loading
