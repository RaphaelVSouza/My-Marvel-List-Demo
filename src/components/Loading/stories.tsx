import { Story, Meta } from '@storybook/react'
import Loading from '.'

export default {
  title: 'Loading',
  component: Loading,
} as Meta

export const Default: Story = () => <Loading />
