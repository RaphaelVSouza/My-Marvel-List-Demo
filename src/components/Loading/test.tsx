import { render, screen } from '@testing-library/react'

import Loading from '.'

describe('<Loading />', () => {
  it('should render the loading', () => {
    const { container } = render(<Loading />)

    expect(
      screen.getByAltText('a circle spinning representing a loading')
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
