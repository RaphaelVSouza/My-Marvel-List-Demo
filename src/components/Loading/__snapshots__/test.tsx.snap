// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the loading 1`] = `
<div
  class="sc-bdnxRM bSTVpm"
>
  <img
    alt="a circle spinning representing a loading"
    class="sc-gtsrHT dLZMpi"
    src="/loading-icon.svg"
  />
</div>
`;
